debezium-reactive/src/main/java/io/debezium/reactive/DebeziumEvent.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.reactive;
+
+import org.apache.kafka.connect.source.SourceRecord;
+
+/**
+ * A class that encapsulates {@link SourceRecord} and associates the {@link #complete()} operation with it.
+ * It is expected that the consumer will receive the event from the pipeline and when the event is
+ * durably processed it will signal this fact back to engine using {@link #complete()} call.
```

Comment:
So what is the purpose of `complete()` exactly? It seems quite "un-reactive" on first thought.

Reply:
This is a hard to answer. We have no way how to detect if the processing of event was completed successfully or not. So we do not include this mechanis it means that the event offset can be committed albeit the consumer will in fact not process it. I can drop it completely but then when pipeline fails the change will not be resent.

Reply:
So why is it you think this should be handled differently from offset management with the existing embedded engine? I.e. can't we leave this as a matter to the configured offset commit strategy, which may commit after every message, once a minute etc.

Of course offsets should only be committed once the handler has returned. Is this the issue perhaps?

Reply:
Exactly - with handler as is for embedded you know if the handler completed sucessfully or thrown an exception - so you commit only in case of the former. This is not possible with RxJava so esentially there are three options
1. Require explicit completion as it is now. The completion is done at the end of pipeline where the application confirms it processed an event
2. Send the event downstream in the pipeline and do not care about the result, it is someone else problem - I am not sure I like this solution
3. Use http://reactivex.io/RxJava/javadoc/io/reactivex/plugins/RxJavaPlugins.html#setErrorHandler-io.reactivex.functions.Consumer- but this is global setting for the whole JVM so I do not think this is viable